### DISCUSSION TITLE
Swell & Sanity.io integration

### DISCUSSION THREAD

**Discussion by studiodrimmel** [swellstores/community#111] [Help] [answered]
Hi All,

I was wondering if anybody has used the combination of Swell with Sanity.io and how you integrated swell within sanity.  I'm looking for  a way to select products from Swell from within Sanity Studio

Example of the desired behaviour:
https://www.velobici.cc/
At the bottom of the page is a section: "Our staff picks"

For example: 
what if I would make a Sanity schema containing a featured products section. Do you somehow import swell products into the sanity studio and select the right swell products from within sanity studio. Or do you guys extend the product model within the swell dashboard with something like a is featured checkbox, allowing it to be toggled as featured product and just fetch those in the component of the desired frontend framework?

Last option is the least desired for me, because if I'd like to make a blogpost I'd like to select a product from within the blogpost in Sanity studio.

Looking for some good integration ideas for the following stack:
Sveltekit
Swell
Sanity

Would love to hear if you guys have a better idea for this

kind regards

**Comment by maarten2424** [comment, marked as answer]
Hey @studiodrimmel,
For sure an integration with Sanity is possible, though we haven't released an official plugin (yet). You could create a custom Sanity input field, where you can query individual products using [swell-js](https://github.com/swellstores/swell-js/). You can return a JSON containing the product data from Swell. Keep in mind data synchronisation; for example a blogpost, you might just need product information, but if you want actual stock and pricing information (or any information that is likely to change) you'll need to refetch this information to be up-to-date in your stack.

I hope this helps you in the right direction!

**Reply by maarten2424** [reply]
Some references:
https://www.sanity.io/docs/custom-input-widgets
https://www.sanity.io/guides/your-first-input-component-for-sanity-studio-v3

**Reply by studiodrimmel** [reply]
Yeah, was thinking in the same direction so I started building the following plugin:

- a custom product type to be used in a sanity schema (array of [{ type: 'product' }]) it would contain just a product name and id for now
- custom input which fetches all products from swell and being able to select a product which would push it to the array

Think this would be good a MVP for now
